File: Algorithm/String/Solution/진법 변환 -douv/radix.c
#include <stdio.h>

int main(void)
{
    int d, r;
    scanf("%d %d", &d, &r);

    char result[14]; // 2^14 = 16384
    int cnt = 0;

    do
    {
        int mod = d % r;
        result[cnt] = mod < 10 ? mod + '0' : (mod - 10) + 'A';
        d /= r;
        cnt++;
    } while (d != 0);

    for (int i = cnt - 1; i >= 0; --i)
    {
        printf("%c", result[i]);
    }

    return 0;
}

File: Algorithm/String/Solution/진법 변환 -douv/radix.py
d, r = map(int, input().split())

cnt = 0
result = []

while True:
    mod = d % r
    result.append(chr(mod + ord('0')) if mod <
                  10 else chr(mod - 10 + ord('A')))
    d //= r
    cnt += 1

    if d == 0:
        break

print(''.join(result)[::-1])
